source/common/router/router_ratelimit.h

```diff
@@ -86,11 +87,18 @@ class RequestHeadersAction : public RateLimit::DescriptorProducer {
  */
 class RemoteAddressAction : public RateLimit::DescriptorProducer {
 public:
+  RemoteAddressAction(const envoy::config::route::v3::RateLimit::Action::RemoteAddress& action)
+      : prefix_len_(PROTOBUF_GET_WRAPPED_OR_DEFAULT(action, prefix_len, MAX_PREFIX_LEN)) {}
+
   // Ratelimit::DescriptorProducer
   bool populateDescriptor(RateLimit::DescriptorEntry& descriptor_entry,
                           const std::string& local_service_cluster,
                           const Http::RequestHeaderMap& headers,
                           const StreamInfo::StreamInfo& info) const override;
+
+private:
+  const uint32_t prefix_len_;
+  static const uint32_t MAX_PREFIX_LEN = 32;
```

Comment:
You set the limit on the API is less than 128, but there is 32. https://github.com/envoyproxy/envoy/pull/20538/files#diff-c1e4a3798005eb4f69d0ea87905f7117810f3c2ef2fd3e06df2d75f3724b5fd7R1777

I guess you should also support IPv6 right?